### PR TITLE
Add setting to specify shellArgs for the automationShell

### LINKED ISSUE #96732: Add setting to specify shellArgs for the automationShell
We can set a separate shell for tasks/debug (using `terminal.integrated.automationShell.windows`) which is a useful feature, but there doesn't seem to be a way to specify shellArgs for the automationShell (and only the automationShell), aka a setting like "terminal.integrated.automationShell.shellArgs.windows"

> NOTE: This is related to https://github.com/microsoft/vscode/issues/90073, but that discusses existing behavior about the current shellArgs setting being ignored. I have no problem with that behavior, I'm asking for a _new feature_ to specify automation-specific shellArgs

As background, I work on the Azure Functions extension and we've had several users reports problems while debugging because they like to use `Set-Location` in their PowerShell profile (Personally I wouldn't do that, but 🤷‍♂️). However, that breaks tasks/debug because the PowerShell profile changes the cwd to the wrong folder. If there was a setting `terminal.integrated.automationShell.shellArgs.windows`, they could add `-NoProfile` as an arg specific to the automation shell and keep `Set-Location` in their profile for non-automation scenarios.

Original issues on the Functions extension: https://github.com/microsoft/vscode-azurefunctions/issues/1175, https://github.com/microsoft/vscode-azurefunctions/issues/1260, https://github.com/Azure/azure-functions-core-tools/issues/1945

### PR DESCRIPTION
**Pull request by alexr00**
Fixes #96732

This is mostly taken from b697be5e975d519a7d2479c7ed1d493b72aefa63.

### CODE REVIEW

File: src/vs/platform/terminal/common/terminalPlatformConfiguration.ts [hunk 1]
```diff
@@ -107,6 +107,36 @@ const terminalPlatformConfiguration: IConfigurationNode = {
 			type: ['string', 'null'],
 			default: null
 		},
+		[TerminalSettingId.AutomationProfileLinux]: {
```
Comment (Tyriar) [inline src/vs/platform/terminal/common/terminalPlatformConfiguration.ts:110]: Should we add `markdownDeprecationMessage` to the old settings?

File: src/vs/workbench/contrib/terminal/browser/terminalProfileResolverService.ts [hunk 2]
```diff
@@ -179,6 +180,15 @@ export abstract class BaseTerminalProfileResolverService implements ITerminalPro
...
 	}
 
+	private _setIconForAutomation(options: IShellLaunchConfigResolveOptions, profile: ITerminalProfile): ITerminalProfile {
+		if (options.allowAutomationShell) {
+			const profileClone = deepClone(profile);
+			profileClone.icon = Codicon.tools;
+			return profileClone;
+		}
+		return profile;
+	}
```
Comment (Tyriar) [inline src/vs/workbench/contrib/terminal/browser/terminalProfileResolverService.ts:190]: Can we just set the icon explicitly inside tasks? Automation profiles are also used by debug and it's not that nice with all these icon special cases. Would this work?

```ts
icon: defaultProfile.icon || Codicon.tools,
```

**Review by Tyriar — CHANGES_REQUESTED** [review]

File: src/vs/workbench/contrib/terminal/browser/terminalProfileResolverService.ts (hunk 2 above)
Comment (alexr00) [reply src/vs/workbench/contrib/terminal/browser/terminalProfileResolverService.ts:190]: I started with that, but the default profile often already has an icon so the `tools` icon will never get used.

**Comment by TSMMark** [comment]
I believe all the PR feedback was addressed, would love to see this released soon to solve the regression that was introduced recently where, to my knowledge, there is no way to have tasks login or use any shell profile.

**Comment by alexr00** [comment]
@Tyriar let me know what else I can do for this PR! It would be great to get the old settings deprecated in favor of the new/better profiles.

**Review by Tyriar — APPROVED** [review]
Sorry about the delay, code looks good on my end if it works 🙂 

Will let you merge as we're approaching the end of the iteration.

**Pull request merged**